exporter/datadogexporter/internal/metrics/utils.go

```diff
@@ -101,13 +101,17 @@ func ProcessMetrics(ms []datadog.Metric) {
 // - hostmetrics receiver metrics (since they clash with Datadog Agent system check) and
 // - running metrics
 func shouldPrepend(name string) bool {
-	namespaces := [...]string{"system.", "process."}
-	for _, ns := range namespaces {
-		if strings.HasPrefix(name, ns) {
-			return true
+	if !strings.HasPrefix(name, "system.") &&
+		!strings.HasPrefix(name, "process.") {
+		return false
+	}
+	for _, t := range metricTransforms {
```

Comment:
Can we have map to datadog infra metrics, so that we can avoid loop ?

Reply:
Good point. I wanted to avoid duplicating their names, to make sure that if anything new gets added to the transform config, it gets picked up here. But I could generate a map on `init`. Is it worth the effort though? This will only get called on occasion for `system.*` or `process.*` metrics. What do you think?

Reply:
Done.